src/VisualStudio/Core/Def/Implementation/EditAndContinue/ManagedEditAndContinueLanguageService.cs

```diff
@@ -87,43 +87,35 @@ public async Task EnterBreakStateAsync(CancellationToken cancellationToken)
 
             try
             {
-                await _proxy.StartEditSessionAsync(cancellationToken).ConfigureAwait(false);
+                await _proxy.BreakStateEnteredAsync(_diagnosticService, cancellationToken).ConfigureAwait(false);
             }
             catch (Exception e) when (FatalError.ReportAndCatchUnlessCanceled(e))
             {
                 _disabled = true;
+                return;
             }
 
             _activeStatementTrackingService.StartTracking();
         }
 
-        public async Task ExitBreakStateAsync(CancellationToken cancellationToken)
+        public Task ExitBreakStateAsync(CancellationToken cancellationToken)
```

Comment:
Why does this still exist? Isn't this just part of CommitSolutionUpdate now?

Reply:
No. This is still called by the debugger and signals end of break state. That's now different from end of edit session.